docs/DeveloperGuide.adoc

```diff
@@ -729,9 +729,59 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 
 |`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
 
+|`* * *` |user |find a person by tags |locate people easily
+
+|`* * *` |user |sort people by name |locate people easily
+
+|`* * *` |user |set a list of favourite people|
+
+|`* * *` |user |see a list of frequently accessed contacts|
+
+|`* * *` |user |get confirmation before I edit or delete contacts|
+
+|`* * *` |user |undo my actions |fix my mistakes
+
+|`* * *` |user |add multiple numbers to a contact |separate their mobile and home numbers
+
+|`* * *` |user |password protect my addressBook |protect my contacts' information
+
+|`* * *` |user |call/message directly from my address book |reach them easily
+
 |`* *` |user |hide link:#private-contact-detail[private contact details] by default |minimize chance of someone else seeing them by accident
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
+|`* *` |user |customise the look of the address book |
+
+|`* *` |user |back up my address book data online |I can access my address book data from other places
+
+|`* *` |user with many friends |autoComplete my searches |
+
+|`* *` |user |share my friend's contact with others |
+
+|`* *` |user |delete all people with a tag |clean up addressBook faster
+
+|`* *` |lazy user |addressBook to auto delete contacts after a certain date |
+
+|`* *` |user |redo an action |redo an undone action
+
+|`* *` |expert user |have separate address books |can separate my work and personal contacts
+
+|`* *` |expert user |customise shorthand commands |so that I can search for contacts faster
+
+|`* *` |user |a calendar to sync my events |can plan them more efficiently
+
+|`* *` |user |know which method of communication my contacts prefers |use their favourite form of communication
+
+|`* *` |user |see most recently accessed contacts |so that I can access them faster
+
+|`*` |user |add profile pictures to contacts |find my friends easily
+
+|`*` |user |set the colour of the tags |can view the tags easily
+
+|`*` |user |see which contacts knows each other |can communicate with them better
```

Comment:
Perhaps the `So that I can` field can be elaborated further as it is rather ambiguous at this point.